core/templates/dev/head/domain/exploration/QuestionListObjectFactorySpec.js

```diff
@@ -158,6 +158,16 @@ describe('Question List Object Factory', function() {
     ).toBe(true);
   });
 
+  it('should remove the given question if it exists', function() {
+    var questionToDelete = sampleQuestions[1];
+    questionList.removeQuestion(questionToDelete);
+    expect(questionList.getQuestionCount())
+      .toEqual(sampleQuestions.length - 1);
+    expect(function() {
+      questionList.getBindableQuestion(questionToDelete.getStateName());
```

Comment:
This is not quite what I meant, by the way. I simply mean verifying the state name of each question in the list,. e.g.:

    expect(questionList.getQuestionCount()).toEqual(1);
    expect(questionList[0].getStateName()).toEqual(sampleQuestions[0].getStateName());
 
so that you basically know what exactly the final list contains.